clang/test/Driver/sycl-offload-static-lib.cpp

```diff
@@ -150,3 +150,15 @@
 // RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl -foffload-static-lib= -c %s 2>&1 \
 // RUN:   | FileCheck %s -check-prefixes=FOFFLOAD_STATIC_LIB_NOVALUE
 // FOFFLOAD_STATIC_LIB_NOVALUE: warning: argument unused during compilation: '-foffload-static-lib='
+
+/// Use of a static archive with various targets should compile and unbundle
+// RUN: touch %t.a
+// RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl %t.a -### 2>&1 \
+// RUN:  | FileCheck %s -check-prefixes=STATIC_ARCHIVE_NO_UNBUNDLE
+// RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_gen-unknown-unknown-sycldevice %t.a -### 2>&1 \
+// RUN:  | FileCheck %s -check-prefixes=STATIC_ARCHIVE_NO_UNBUNDLE
+// RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_fpga-unknown-unknown-sycldevice %t.a -### 2>&1 \
+// RUN:  | FileCheck %s -check-prefixes=STATIC_ARCHIVE_NO_UNBUNDLE
+// RUN: %clangxx -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_x86_64-unknown-unknown-sycldevice %t.a -### 2>&1 \
+// RUN:  | FileCheck %s -check-prefixes=STATIC_ARCHIVE_NO_UNBUNDLE
+// STATIC_ARCHIVE_NO_UNBUNDLE: clang-offload-bundler{{.*}}
```

Comment:
Could `STATIC_ARCHIVE_UNBUNDLE` or `STATIC_ARCHIVE_ALWAYS_UNBUNDLE` end up being a better check prefix? The 'negated' name really didn't parse for me until I figured it's named after the bug and not after the resolution.

Reply:
Yup, that can be updated to better reflect the check.